adagrams/game.py

```diff
@@ -1,11 +1,168 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+
+
+
+
+
 def draw_letters():
-    pass
+
+    letter_pool = [
+        'J', 'K', 'Q', 'X', 'Z', 
+        'B', 'C', 'F', 'H', 'M', 'P', 'V', 'W', 'Y',
+        'G',
+        'D', 'L', 'S', 'U', 
+        'N', 'R', 'T', 
+        'O', 
+        'A', 'I', 
+        'E'
+    ]
+
+    weight = [
+        1, 1, 1, 1, 1, 
+        2, 2, 2, 2, 2, 2, 2, 2, 2, 
+        3,
+        4, 4, 4, 4,
+        6, 6, 6,
+        8, 
+        9, 9, 
+        12
+    ]
+
+    user_hand = []
+
+    for i in range(10): 
+        random_choice = random.choices(letter_pool, weights=weight, k = 1)
+
+        tile = letter_pool.index(random_choice[0])
+        weight[tile] -= 1
+        if weight[tile] == 0: 
+            weight.pop(tile)
+            letter_pool.pop(tile)
+
+        user_hand.append(random_choice[0])
+        
+    return user_hand
+
+
+
+
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    uppercase_word = word.upper()
+    for char in uppercase_word: 
+        if char not in letter_bank or uppercase_word.count(char) > letter_bank.count(char):
+            return False  
+    return True 
```

Comment:
Nice use of `.count()`! I also noticed you returned True no matter what happens because False will only be returned in that conditional is met 👍🏾 You could also use `word.upper()` in your for loop instead of creating another variable in memory 🥳